docker-storage-setup.sh

```diff
@@ -366,6 +368,19 @@ get_configured_thin_pool() {
   done
 }
 
+check_docker_storage_metadata() {
+  local docker_devmapper_meta_dir="$DOCKER_METADATA_DIR/devicemapper/metadata/"
+
+  [ ! -d "$docker_devmapper_meta_dir" ] && return 0
+
+  # Docker seems to be already using devicemapper storage driver. Error out.
+  Error "Docker devicemapper metadata directory $docker_devmapper_meta_dir is present."
+  Error "It seems docker is already having devicemapper metadata for a specific thin pool. Not creating a new thin pool as existing docker metadata will fail to work with new thin pool. Cleanup existing docker metadata and retry."
```

Comment:
Since this is a user-visible error, I'd suggest cleaning up the language:
"Docker has been previously configured for use with devicemapper metadata. Not creating a new thin pool as existing docker metadata will fail to work with it. Manual cleanup is required before this will succeed."